airflow_provider_fivetran/hooks/fivetran.py

```diff
@@ -0,0 +1,242 @@
+                )
+                response.raise_for_status()
+                return response.json()
+            except requests_exceptions.RequestException as e:
+                if not _retryable_error(e):
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException(
+                        f'Response: {e.response.content}, Status Code: {e.response.status_code}'
+                    )
+
+                self._log_request_error(attempt_num, e)
+
+            if attempt_num == self.retry_limit:
+                raise AirflowException(
+                    ('API requests to Fivetran failed {} times. ' + 'Giving up.').format(self.retry_limit)
+                )
+
+            attempt_num += 1
+            sleep(self.retry_delay)
+
+    def _log_request_error(self, attempt_num: int, error: str) -> None:
+        self.log.error('Attempt %s API Request to Fivetran failed with reason: %s', attempt_num, error)
+
+    def check_connector(self, connector_id):
+        # Make sure connector configuration has been completed successfully and is not broken.
+        
+        resp = self._do_api_call(('GET',connector_id))
+        connector_details = resp["data"]
+        URL_LOGS = "https://fivetran.com/dashboard/connectors/{}/{}/logs".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        URL_SETUP = "https://fivetran.com/dashboard/connectors/{}/{}/setup".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        setup_state = connector_details["status"]["setup_state"]
+        if setup_state != "connected":
+            EXC_SETUP: str = (
+                'Fivetran connector "{}" not correctly configured, status: {}; '
+                + "please complete setup at {}"
+            )
+            raise ValueError(
+                EXC_SETUP.format(self.connector_id, setup_state, URL_SETUP)
+            )
+        # We need to know the previous job's completion time to know if the job succeeded or failed
+        succeeded_at = self.parse_timestamp(connector_details["succeeded_at"])
+        failed_at = self.parse_timestamp(connector_details["failed_at"])
+        previous_completed_at = succeeded_at if succeeded_at > failed_at else failed_at
+        # URL for connector logs within the UI
+        log.info(
+            "Connector type: {}, connector schema: {}".format(
+                connector_details["service"], connector_details["schema"]
+            )
+        )
+        log.info("Connectors logs at {}".format(URL_LOGS))
+        return resp
+
+    def set_manual_schedule(self, connector_id):
+        # Set connector to manual sync mode, required to force sync through the API
+        return self._do_api_call(('PATCH',connector_id),json.dumps({"schedule_type": "manual"}))
+
```

Comment:
two line breaks between methods

Reply:
We should filter all files through black -- I'll show you how to set this up when we talk. We can also make this a part of the commit CI process.